admission-webhook/webhook.go

```diff
@@ -392,15 +403,31 @@ func (webhook *webhook) mutateCreateRequest(ctx context.Context, pod *corev1.Pod
 
 	admissionResponse := &admissionV1.AdmissionResponse{Allowed: true}
 
-	if len(patches) != 0 {
-		patchesBytes, err := json.Marshal(patches)
-		if err != nil {
-			return nil, &podAdmissionError{error: fmt.Errorf("unable to marshall patch JSON %v: %v", patches, err), pod: pod, code: http.StatusInternalServerError}
+	if hasGMSA && webhook.config.EnableRandomHostName {
+		// Pods are GMSA related, Env enabled, patch the hostname only if it is empty
+		hostName := pod.Spec.Hostname
+		if hostName == "" {
+			hostName = generateUUID()
+			patches = append(patches, map[string]string{
+				"op":    "add",
+				"path":  "/spec/hostname",
+				"value": hostName,
+			})
+		} else {
+			// Will honor the hostname set in the spec, print out a message
+			logrus.Warnf("hostname is set in spec and will be hornored instead of being randomized")
 		}
 
-		admissionResponse.Patch = patchesBytes
-		patchType := admissionV1.PatchTypeJSONPatch
-		admissionResponse.PatchType = &patchType
+		if len(patches) != 0 {
```

Comment:
this needs to be outside the `if hasGMSA && webhook.config.EnableRandomHostName {`

Reply:
right, thanks for catching this